Storage/rest/hypermedia.h

```diff
@@ -31,39 +31,146 @@ SOFTWARE.
 #include "../../TypeSystem/struct.h"
 #include "../../Blocks/HTTP/api.h"
 
+namespace current {
+namespace storage {
+namespace rest {
+
 CURRENT_STRUCT(HypermediaRESTTopLevel) {
   CURRENT_FIELD(url, std::string);
   CURRENT_FIELD(url_status, std::string);
   CURRENT_FIELD(url_data, (std::map<std::string, std::string>));
   CURRENT_FIELD(up, bool);
+
   CURRENT_CONSTRUCTOR(HypermediaRESTTopLevel)(const std::string& url_prefix = "", bool up = false)
       : url(url_prefix), url_status(url_prefix + "/status"), up(up) {}
 };
 
 CURRENT_STRUCT(HypermediaRESTStatus) {
   CURRENT_FIELD(up, bool);
+
   CURRENT_CONSTRUCTOR(HypermediaRESTStatus)(bool up = false) : up(up) {}
 };
 
-CURRENT_STRUCT(HypermediaRESTContainerResponse) {
+CURRENT_STRUCT(HypermediaRESTError) {
+  using T_DETAILS = std::map<std::string, std::string>;
+  CURRENT_FIELD(error, std::string);
```

Comment:
I've edited my comment:
> This is commonly called "name" or "code" (code is generally a number, but sometimes string).

"ParseJSONError" looks more like a "name", not a "code".
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/name